src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/JpaReviewSchedulePersistenceAdapter.kt

```diff
@@ -71,6 +77,30 @@ class JpaReviewSchedulePersistenceAdapter(
       updatedAtPrison = updateReviewScheduleStatusDto.prisonId
     }
 
-    return reviewScheduleEntityMapper.fromEntityToDomain(reviewScheduleRepository.save(reviewScheduleEntity))
+    val savedReviewScheduleEntity = reviewScheduleRepository.save(reviewScheduleEntity)
+    saveReviewScheduleHistory(savedReviewScheduleEntity)
+    return reviewScheduleEntityMapper.fromEntityToDomain(savedReviewScheduleEntity)
+  }
+
+  fun saveReviewScheduleHistory(reviewScheduleEntity: ReviewScheduleEntity) {
+    with(reviewScheduleEntity) {
+      val historyEntry = ReviewScheduleHistoryEntity(
+        version = reviewScheduleHistoryRepository.findMaxVersionByReviewScheduleReference(reference)
+          ?.plus(1) ?: 1,
```

Comment:
Look up the existing history for this reviewSchedule and increment the version number (if none found then it's the first)